helpdesk_mgmt/tests/test_portal.py

```diff
@@ -0,0 +1,147 @@
+from odoo import http
+from odoo.tests.common import new_test_user
+
+from odoo.addons.base.tests.common import HttpCaseWithUserPortal
+
+
+class TestPortal(HttpCaseWithUserPortal):
+    """Test controllers defined for portal mode.
+    This is mostly for basic coverage; we don't go as far as fully validating
+    HTML produced by our routes.
+    """
+
+    def setUp(self):
+        """Override test set-up to prepare data used in tests."""
+
```

Comment:
Please avoid empty lines inside the method. They are discouraged to be used.

Reply:
Ok I can see in partner_firstname tests we don't add new lines after docstrings ➔ I have removed all NLs after docstrings

Also removed that setUp docstring as it is too usual to need explanation, I can see other tests never document it

The only NLs remaining are those that make sense inside longer tests (such as `test_tickets_2_users`)

I didn't see this mentioned on https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst would be interesting to add I think :)

Reply:
Yeah, docstrings are very useful, specially if the test name is not descriptive enough. About including this in the guidelines, the idea is to avoid to repeat all the Python conventions. Specifically, this one is already pointed in the old PEP8 one: https://peps.python.org/pep-0008/#blank-lines

This was later a bit nuanced, as having empty lines inside the method, even if logical groups, is worse than having a comment line saying what the logical group does, so that's also my preferred option.